### PR TITLE
[xcvrd] Add bitmap support for SFP error event

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
#### Description

Support SFP error bitmap.

#### Motivation and Context

Currently, SONiC use a single value to represent SFP error, however, multiple SFP errors could exist at the same time. This PR is aimed to support it   

#### How Has This Been Tested?

1. New unit test case is added to cover the change
2. Manual test

### CODE REVIEW

File: sonic-xcvrd/tests/test_xcvrd.py [hunk 1]
```diff
@@ -24,13 +24,12 @@
...
 scripts_path = os.path.join(modules_path, "xcvrd")
-helper_file_path = os.path.join(scripts_path, "xcvrd_utilities"+"/y_cable_helper.py")
 sys.path.insert(0, modules_path)
 
 os.environ["XCVRD_UNIT_TESTING"] = "1"
-load_source('y_cable_helper', scripts_path + '/xcvrd_utilities/y_cable_helper.py')
-from y_cable_helper import *
+from xcvrd.xcvrd_utilities.y_cable_helper import *
 from xcvrd.xcvrd import *
+from xcvrd.xcvrd_utilities.sfp_status_helper import *
```
Comment (jleveque) [inline sonic-xcvrd/tests/test_xcvrd.py:32]: Suggest reordering:

```python
from xcvrd.xcvrd import *
from xcvrd.xcvrd_utilities.sfp_status_helper import *
from xcvrd.xcvrd_utilities.y_cable_helper import *
```

File: sonic-xcvrd/tests/test_xcvrd.py [hunk 2]
```diff
@@ -24,13 +24,12 @@
 test_path = os.path.dirname(os.path.abspath(__file__))
 modules_path = os.path.dirname(test_path)
 scripts_path = os.path.join(modules_path, "xcvrd")
-helper_file_path = os.path.join(scripts_path, "xcvrd_utilities"+"/y_cable_helper.py")
 sys.path.insert(0, modules_path)
 
 os.environ["XCVRD_UNIT_TESTING"] = "1"
-load_source('y_cable_helper', scripts_path + '/xcvrd_utilities/y_cable_helper.py')
```
Comment (jleveque) [inline sonic-xcvrd/tests/test_xcvrd.py:31]: With the removal of this line, I believe we should also be able to remove `from imp import load_source` on line 7. I'm surprised LGTM didn't complain about an unused import.

**Review by jleveque — CHANGES_REQUESTED** [review]

**Comment by stephenxs** [comment]
In order to have this PR passed CI, we need to merge Azure/sonic-platform-common#194 and then advance the submodule head for it. It references some definitions like `SFP_ERROR_DESCRIPTION_BLOCKING ` which are defined there.
```
2021-06-11T22:03:53.0398651Z =================================== FAILURES ===================================
2021-06-11T22:03:53.0399461Z _______________ TestXcvrdScript.test_detect_port_in_error_status _______________
2021-06-11T22:03:53.0399857Z 
2021-06-11T22:03:53.0400714Z self = <tests.test_xcvrd.TestXcvrdScript object at 0x7fc5961cea50>
2021-06-11T22:03:53.0401084Z 
2021-06-11T22:03:53.0401598Z     def test_detect_port_in_error_status(self):
2021-06-11T22:03:53.0402206Z         class MockTable:
2021-06-11T22:03:53.0402786Z             def get(self, key):
2021-06-11T22:03:53.0403305Z                 pass
2021-06-11T22:03:53.0403736Z     
2021-06-11T22:03:53.0404180Z         status_tbl = MockTable()
2021-06-11T22:03:53.0405776Z         status_tbl.get = MagicMock(return_value=(True, {'error': 'N/A'}))
2021-06-11T22:03:53.0406569Z >       assert not detect_port_in_error_status(None, status_tbl)
2021-06-11T22:03:53.0406913Z 
2021-06-11T22:03:53.0407370Z tests/test_xcvrd.py:326: 
2021-06-11T22:03:53.0408023Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2021-06-11T22:03:53.0408410Z 
2021-06-11T22:03:53.0408841Z logical_port_name = None
2021-06-11T22:03:53.0409434Z status_tbl = <tests.test_xcvrd.MockTable instance at 0x7fc596c0a320>
2021-06-11T22:03:53.0409817Z 
2021-06-11T22:03:53.0410397Z     def detect_port_in_error_status(logical_port_name, status_tbl):
2021-06-11T22:03:53.0411119Z         rec, fvp = status_tbl.get(logical_port_name)
2021-06-11T22:03:53.0411684Z         if rec:
2021-06-11T22:03:53.0412179Z             status_dict = dict(fvp)
2021-06-11T22:03:53.0413086Z             error = status_dict.get('error')
2021-06-11T22:03:53.0413762Z >           return SfpBase.SFP_ERROR_DESCRIPTION_BLOCKING in error
2021-06-11T22:03:53.0415015Z E           AttributeError: type object 'SfpBase' has no attribute 'SFP_ERROR_DESCRIPTION_BLOCKING'
2021-06-11T22:03:53.0415493Z 
2021-06-11T22:03:53.0416035Z xcvrd/xcvrd_utilities/sfp_status_helper.py:35: AttributeError
2021-06-11T22:03:53.0417208Z ---------- generated xml file: /__w/1/s/sonic-xcvrd/test-results.xml -----------
2021-06-11T22:03:53.0417626Z 
2021-06-11T22:03:53.0418544Z ---------- coverage: platform linux2, python 2.7.16-final-0 ----------
2021-06-11T22:03:53.0419294Z Name                                         Stmts   Miss  Cover
2021-06-11T22:03:53.0420322Z ----------------------------------------------------------------
2021-06-11T22:03:53.0421049Z xcvrd/__init__.py                                0      0   100%
2021-06-11T22:03:53.0421738Z xcvrd/xcvrd.py                                 824    566    31%
2021-06-11T22:03:53.0422892Z xcvrd/xcvrd_utilities/__init__.py                0      0   100%
2021-06-11T22:03:53.0423781Z xcvrd/xcvrd_utilities/sfp_status_helper.py      25      9    64%
2021-06-11T22:03:53.0424490Z xcvrd/xcvrd_utilities/y_cable_helper.py        650    584    10%
2021-06-11T22:03:53.0425562Z ----------------------------------------------------------------
2021-06-11T22:03:53.0426286Z TOTAL                                         1499   1159    23%
2021-06-11T22:03:53.0426906Z Coverage HTML written to dir htmlcov
2021-06-11T22:03:53.0427487Z Coverage XML written to file coverage.xml
2021-06-11T22:03:53.0427772Z 
2021-06-11T22:03:53.0428338Z ===================== 1 failed, 11 passed in 0.60 seconds ======================
```

**Comment by stephenxs** [comment]
Submodule head of sonic-platform-common is advanced in Azure/sonic-buildimage#7868

**Comment by stephenxs** [comment]
@jleveque Can you trigger the test rerun as the dependent PR has been merged. Thanks

**Review by jleveque — APPROVED** [review]

**Pull request merged**